float32[] angles
float32 range_max
float32 range_min
